features/proposed/FxA-70-kpi-dashboards/README.md

```diff
@@ -0,0 +1,414 @@
+  the number of distinct user ids
+  for which we have at least one activity event in our logs
+  during a given UTC day.
+
+* "Active monthly users" or "AMU" on a given day means
+  the number of distinct user ids
+  for which we have at least one activity event in our logs
+  during the previous 30 UTC days.
+
+* "Engagement ratio" on a given day means
+  the mean ADU over the previous 7 days,
+  divided by the AMU for that day,
+  expressed as a percentage.
+
+* A userid is a "multi-device user" on a given day
+  if their account had an active device on that UTC day,
+  and had an active device with a different device id
+  in the previous five UTC days.
+
+* "Multi-Device ADU" means
+  the number of distinct user ids
+  which counted as multi-device users on a given UTC day.
+
+* "Multi-Device AMU" on a given day means
+  the number of distinct user ids
+  which counted as multi-device users
+  at any time during the previous 30 UTC days.
+
+* A device is "ready to sync"
+  when it has registered a device metadata record
+  with the FxA auth backend,
+  and it has successfully fetched encryption keys
+  from the server.
+
+* "Time to device connection" means the time elapsed,
+  in seconds,
+  between the device first requesting connection UI
+  from the FxA front-end servers,
+  and the device being ready to sync.
+
+The definition of a "multi-device user" here
+is somewhat arbitrary,
+but is designed to capture
+users who are actively engaged in a "multi-device experience".
+Only counting users with
+multiple devices active on a single day
+would seem too restrictive to
+encompass more casual users,
+while counting devices used weeks apart
+would not seem to capture what's really meant
+by "multi-device experience".
+
+### Data Capture and Storage
+
+We will use AWS Redshift as the canonical store
+for all our KPI-related metrics data,
+and will extend our existing "activity events" metrics stream
+to track all the information required
+to generate these KPI dashboards.
+All events will be generated server-side.
+
+All the events generated
+by a particular user interaction with FxA
+will be grouped into a "sequence"
+via a shared random identifier.
+This will enable us to correlate events
+across time and across services.
+For example,
+a sequence might consist of individual events for
+the user initiating a connection attempt,
+successfully creating an account,
+and successfully connecting their device.
+
+All generated events will be stored
+into a single table
+with the following core event-data columns:
+
+* **timestamp**:  UTC timestamp at which the event occurred
+* **type**:  event type; see below
+* **uid**:  user id that generated the event, if known
+* **deviceId**:  device id that generated the event, if known
+* **seqId**:  a unique id for a particular user interaction sequence spanning multiple events
+* **seqTime**:  time in milliseconds between the beginning of the sequence and this event
```

Comment:
I found it really useful for myself to provide a row of fake data to visualize this better, see example : https://mana.mozilla.org/wiki/display/CLOUDSERVICES/Retention+Metrics up to you if you want to include it or not.